Commons.Music.Midi.DesktopShared/winmm/WinMMNatives.cs

```diff
@@ -40,18 +41,23 @@ struct MidiHdr
 		public IntPtr Next; // of MidiHdr
 		public IntPtr Reserved;
 		public int Offset;
-		public IntPtr Reserved2;
-	}
 
-	public enum MidiInOpenFlags
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
+        public int[] reservedArray;
```

Comment:
What do you mean by "copy by value semantics"?

We can go with StructLayoutAttribute Size if you want, I just want it to be the correct size so that it works.

Reply:
Usually value types such as structs don't need special object tracking in stack memory for GC but it becomes required if a value type object references a managed object (such as an array) and it will give some additional cost on GC (it's tiny but I'd avoid if there is no reason not to).

But instead of specifying explicit size at struct level, I'm thinking to use this now: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/unsafe-code-pointers/fixed-size-buffers

Reply:
Interesting, didn't know that. I'll change it to fixed then?

Reply:
Ah, just leave it as you like. I will make changes.

Reply:
Hmm, UnmanagedType.ByValArray makes the array a fixed size, by value, non-GC type already (otherwise it wouldn't work with C). So I'm guessing it's actually fine as it is. But if you want to change it go ahead.

Reply:
Marshaling and GC lookup are different issues.

Reply:
Ah, yes, of course.